### ISSUE TITLE
Improve ensuring that a resource group exists

### ISSUE THREAD

**Issue by carolynvs** [getporter/az-mixin#32]
I just saw @squillace 's hack to ensure that a resource group exists. Let's improve the mixin so that this doesn't require dropping into the exec mixin.

It's important that mixins should gracefully handle resources that already exist (during install) or conversely resources that have already been deleted (during uninstall).

Ralph, can you provide the command that you wanted to run with the mixin and also the bash hack?

**Comment by squillace** [comment]
here's an example: https://github.com/squillace/cnab-aks-winlin/blob/main/upsertGroup.sh. The problem is that arm or the azure cli errors out if the group already exists, which is strange as all the other operations are a no-op in that case.

**Comment by carolynvs** [comment]
What do you think about this? It adds a declarative form for resource groups that removes the need for any workarounds.

```
# Creates the group if it doesn't exist
install:
  az:
    group: NAME
    location: LOCATION

# Removes the group if it exists
uninstall:
  az:
    group: NAME
    location: LOCATION
```

**Comment by carolynvs** [comment]
This was fixed in #35
